File: UVa/cpe_study/Solo/week7/1583/sol_2.cpp
#include <bits/stdc++.h>
#define endl "\n"
#define IO ios::sync_with_stdio(0), cin.tie(0), cout.tie(0)
#define MAXN maxn
#define MOD modn

int digGen[100005];

using namespace std;

void dig(){
    int sum, tmp;
    for(int i=100000; i>=1; i--) {
        sum = i; tmp = i;
        while(tmp > 0) {
            sum += tmp%10;
            tmp /= 10;
        }
        digGen[sum] = i;
    }
}

signed main() {
    dig();
    int N, targ;
    cin>>N;
    while (N--){
        cin>>targ;
        cout<<digGen[targ]<<endl;
    }
    return 0;
}

File: uva/other/1583/sol_2.py
dig = {}

def genDig():
    for i in range(100000):
        dig[i] = 0
    for i in range(100000, 0, -1):
        sum = i; tmp = i
        while tmp > 0:
            sum += tmp%10
            tmp //= 10
        dig[sum] = i

if __name__ == '__main__':
    genDig()
    n = int(input())
    while n:
        n -= 1
        targ = int(input())
        if (dig[targ]):
            print(dig[targ])
        else:
            print(0)